runtime/android/core_internal/src/org/xwalk/core/internal/SwipeRefreshHandler.java

```diff
@@ -0,0 +1,185 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Copyright (c) 2016 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package org.xwalk.core.internal;
+
+import android.content.Context;
+import android.view.ViewGroup.LayoutParams;
+
+import org.chromium.base.annotations.JNINamespace;
+import org.chromium.base.metrics.RecordUserAction;
```

Comment:
I'm not sure it's worth duplicating even the tracings and metrics since they don't provide anything useful to us.